packages/smooth_app/lib/helpers/picture_capture_helper.dart

```diff
@@ -1,39 +1,105 @@
+import 'dart:io';
+import 'dart:math';
 import 'package:flutter/material.dart';
 import 'package:flutter_gen/gen_l10n/app_localizations.dart';
 import 'package:openfoodfacts/openfoodfacts.dart';
 import 'package:provider/provider.dart';
 import 'package:smooth_app/data_models/continuous_scan_model.dart';
 import 'package:smooth_app/database/product_query.dart';
-import 'package:smooth_app/generic_lib/loading_dialog.dart';
+import 'package:workmanager/workmanager.dart';
+
+@pragma(
+    'vm:entry-point') // Mandatory if the App is obfuscated or using Flutter 3.1+
+void callbackDispatcher() {
+  Workmanager().executeTask(
+    (String task, Map<String, dynamic>? inputData) async {
+      // make a counter with task as key as it is unique for each task
+      final int counter = inputData!['counter'] as int;
+      // if task is greate than 6 , that means it has been executed 7 times
+      if (counter > 6) {
+        // returns true to let platform know that the task is completed
+        final File file = File(inputData['imageUri'].toString());
+        file.delete();
+        return Future<bool>.value(true);
```

Comment:
Since your method is `async`, just `return true`

Reply:
marking these as resolved cuz done in the latest commit